### ISSUE TITLE
How to choose the far_target

### ISSUE THREAD

**Issue by LiuNull** [davidsandberg/facenet#625]
I was wondering if the far_target is chosen in a random way or someway else. I am currently running FaceNet with my own face dataset, but I got a bad validation rate (14%) when I use 1e-3 as the far_target, and in this case my threshold was 0.25, it is strange. I though the threshold should be 1.021.

**Comment by Balkandaene** [comment]
Hi Liu,

how is your dataset composed in terms of ethnicity? In my experience, the pre-trained MS-celeb model is quite bad at distinguishing Asian people. This bummer is likely due to a bias for Caucasian people in the training set.  Similar problems with Chinese faces have been reported for a huge vendor of mobile phones...

Using the **"standard"** **threshold of **1.021**** from LFW, I got a **FAR of 15%** (0.15) in my testset of **Asian faces**. If I would now specify a **FAR target of 0.1%** (1e-3) for this dataset, the threshold would certainly decrease **way** **below 1.021** and accordingly the **validation rate drop**. But I still have to say that I am kinda surprised how strict the threshold is and how bad your validation rate is. 

In short, I guess the threshold really depends on your dataset and how well the model was trained for it. If the composition in your dataset is very different, the low validation rate maybe (at least partially) be explained by this factor. You may want to consider to retrain your model with relevant images/faces to enhance your results. @LiuNull

**Comment by LiuNull** [comment]
@Balkandaene 
Thank you for answering my question.

My dataset contains Asian faces. And I use the pre-trained **CASIA-WebFace** model for another training. My accuracy has risen by **three percentage points** , it is a great improvement because you said the MS-celeb model is bad at distinguishing Asian people. However the validation rate is still **low**(13%). 

And now I have a new question, the threshold is **0.91** when I calculate the accuracy, but it turned **0. 29** when I calculate the validation rate. I am quite confused about it. I think it is the far_target that caused the difference. But I don't know how to choose the far_target to get a ideal validation rate. 

Hope to get your reply.

**Comment by Balkandaene** [comment]
@LiuNull 

Great if my answer was of some help.

I actually did not check the CASIA-model on Asians faces. Therefore, I cannot say whether it performs better than the MS_Celeb model, or whether an additional training of data from CASIA-WebFace can enhance the MS-Celeb model.

The FAR-target and the validation rate are closely related/intermingled. You can get a validation curve by changing the FAR-target to higher values: e.g. you could increase the FAR-target from 0.1% (as is now), to 0.2 %, to 0.5 % and to 1.0 % (and so on). By doing so, you will get an increasing validation rate. It is your specific project/task which determines what FAR can be acceptable (e.g. in security related issues you do not want to risk anything and the FAR should be low; in other settings you may tolerate more false positives...)

I think it would be very useful if you take a look at **figure 5 of the ****original Facenet paper by Google** (https://arxiv.org/abs/1503.03832). In this figure, they did exactly what I have just described to you: changing the FAR-target and see the resulting validation rate. Moreover, they imply that the steep drop of the validation rate at strict FARs (FAR < 1e-5) could (at least partially) be due to labeling noise. Therefore, you should critically check whether your dataset is sufficiently clean: It could be that your bad validation is not only related to a poorly pre-trained model for Asian faces, but also to labeling noise in your own dataset.
